ndcube/ndcube.py

```diff
@@ -1122,6 +1123,8 @@ def my_propagate(uncertainty, data, mask, **kwargs):
         """
         # Sanitize input.
         new_unit = new_unit or self.unit
+        if isinstance(bin_shape, u.Quantity):
+            bin_shape = bin_shape.to_value()
```

Comment:
Surely we only want to allow units of pixels? or maybe pixels and dimensionless?

Reply:
Up to you, I can check its pixel and error if not and then remove the unit.